src/library_syscall.js

```diff
@@ -1830,8 +1824,20 @@ if (SYSCALL_DEBUG) {
 var WASI_SYSCALLS = set([
   'fd_write',
   'fd_close',
+  'fd_seek',
 ]);
 
+// Fallback for cases where the wasi_unstable.name prefixing fails,
+// and we have the full name from C. This happens in fastcomp (which
+// lacks the attribute to set the import module and base names) and
+// in LTO mode (as bitcode does not preserve them).
+
+if (!WASM_BACKEND || !WASM_OBJECT_FILES) {
```

Comment:
This might be a side question or one for @sbc100 but do you know if (WASM_BACKEND && !WASM_OBJECT_FILES) uses the emscripten driver code to do the bitcode linking, or the one in clang/lld? IIUC the upstream backend bitcode does preserve the prefixing, because how else would the attribute propagate from clang to the backend? I guess I just don't understand why (WASM_BACKEND && !WASM_OBJECT_FILES)  doesn't work here.

Reply:
This is a bug in wasm-ld where linking with `-r/--relocatable` doesn't preserve these attributes.  I'm working on a fix at the moment.